doc/releases/changelog-dev.md

```diff
@@ -430,6 +430,8 @@ such as `shots`, `rng` and `prng_key`.
 
 <h3>Labs 🧪</h3>
 
+<h4>Resource Estimation Infrastructure</h4>
```

Comment:
```suggestion
<h4>Resource estimation</h4>
```

Reply:
Do we also need "Other Improvements" => "Other improvements" on [line 341](https://github.com/PennyLaneAI/pennylane/pull/6744/files#diff-e6ab4fddf160607572464ae0724adc6cf2cb437959a8f238382eb0d0befc92f7R341)?

Reply:
It was like that in the 0.39 release notes, but not 0.38. I think it makes sense to leave it as is.